services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker_core.py

```diff
@@ -46,11 +52,41 @@ def _prometheus_sync_client_custom_query(
     return data
 
 
+def _to_int_or_none(value) -> int | None:
+    return int(value) if value else None
+
+
+def _build_cache_key_user_id(fct, *args):
+    return f"{fct.__name__}_{args[1]}"
+
+
+def _build_cache_key_project_and_node_id(fct, *args):
+    return f"{fct.__name__}_{args[1]}_{args[2]}"
+
+
+@cached(ttl=_TTL, key_builder=_build_cache_key_user_id)
+async def _get_user_email(osparc_repo: OSparcRepository, user_id: int) -> str | None:
+    user_email: str | None = await osparc_repo.get_user_email(user_id)
+    return user_email
+
+
+@cached(ttl=_TTL, key_builder=_build_cache_key_project_and_node_id)
+async def _get_project_and_node_names(
+    osparc_repo: OSparcRepository, project_uuid: ProjectID, node_uuid: NodeID
+) -> tuple[str | None, str | None]:
+    output = await osparc_repo.get_project_name_and_workbench(project_uuid)
+    if output:
+        project_name, project_workbench = output
+        return (project_name, project_workbench[f"{node_uuid}"].get("label"))
+    return (None, None)
```

Comment:
This might be an issue. What happens, if I start a node, then add a new one after 5 minutes? Have you considered this situation? Would the data not be out of date?
I would not chose to cache the entire workbench here!

Reply:
Thanks good catch! I do not cache the whole workbench, just the specific node_uuid! but still you are right, i will reduce it to 5 mins, should be sufficient for now :+1: